src/__tests__/App.test.tsx

```diff
@@ -1,13 +1,19 @@
 import React from "react";
-import ReactDOM from "react-dom";
+import { createRoot } from "react-dom/client";
 import App from "../App";
 import { cleanup } from "@testing-library/react";
+import { BrowserRouter as Router } from "react-router-dom";
 
 afterEach(cleanup);
 
 describe("App", () => {
   it("renders without crashing", () => {
     const div = document.createElement("div");
-    ReactDOM.render(<App />, div);
+    const root = createRoot(div!);
```

Comment:
To get rid of this typescript error, we can just check if div is truthy

```js
    if (div) {
      const root = createRoot(div);
      root.render(
        <Router>
          <App />
        </Router>
      );
    }
```

Reply:
we should still silence the error though.
If you want to keep the original version, then you will need to add an es lint disable line 

```
/* eslint-disable  @typescript-eslint/no-non-null-assertion */
```
The welcome page test will need the same fix